src/main/java/ratpack/zipkin/internal/ZipkinHttpClientImpl.java

```diff
@@ -81,18 +75,16 @@ public void close() {
 
     @Override
     public Promise<ReceivedResponse> request(URI uri, Action<? super RequestSpec> action) {
-        return delegate.request(uri, tracedRequestAction(action))
-                .wiretap(receivedResponseResult -> responseInterceptor
-                        .handle(responseAdapterFactory.createAdapter(receivedResponseResult.getValue()))
-                );
+        final ExecuteWithSpan execution = new ExecuteWithSpan(tracer);
+        return delegate.request(uri, execution.execute(action))
```

Comment:
it might be useful to eagerly provision a span, then wrap delegate.request with `try(SpanInScope ws = tracer.withSpanInScope(span))` as that would allow anything invoking downstream see it (potentially helpful if SLF4J MDC is in use). You can lazy start that span in the execution thing..